core/server/common/src/main/java/alluxio/Registry.java

```diff
@@ -36,47 +36,50 @@
 import javax.annotation.concurrent.ThreadSafe;
 
 /**
- * Class for registering individual masters that make up the Alluxio master service. The intended
- * use of this class is for the Alluxio master service to register the individual masters using an
+ * Class for registering individual {@link Server}s that run within an Alluxio process. The intended
+ * use of this class is for the Alluxio process to register the individual {@link Server}s using an
  * instance of this class. The registry is passed as an argument to constructors of individual
- * masters who are expected to add themselves to the registry.
+ * {@link Server}s who are expected to add themselves to the registry.
  *
  * The reason for using an instance as opposed to a static class is to enable dependency
  * injection in tests.
+ *
+ * @param <T> the type of the {@link Server}
  */
 @ThreadSafe
-public final class MasterRegistry {
+public final class Registry<T extends Server> {
```

Comment:
We can get argument type safety by changing this to 

`<T extends Server<OPT>, OPT>`

and then creating subclasses to bind the server and argument types together:

```java
public class MasterRegistry extends Registry<Master, Boolean> {}

public class WorkerRegistry extends Registry<Worker, WorkerNetAddress> {}
```

Then use the types `MasterRegistry` and `WorkerRegistry`